nixos/modules/services/x11/desktop-managers/plasma5.nix

```diff
@@ -365,6 +373,26 @@ in
       security.pam.services.lightdm.enableKwallet = true;
       security.pam.services.sddm.enableKwallet = true;
 
+      systemd.user.services = mkIf cfg.runUsingSystemd {
+        plasma-early-setup = {
+          description = "Early Plasma setup";
+          wantedBy = [ "graphical-session-pre.target" ];
+          serviceConfig.Type = "oneshot";
+          script = activationScript;
+        };
+
+        plasma-run-with-systemd = {
+          description = "Run KDE Plasma via systemd";
+          wantedBy = [ "basic.target" ];
+          serviceConfig.Type = "oneshot";
+          script = ''
+            ${set_XDG_CONFIG_HOME}
+
+            echo -e "[General]\nsystemdBoot=true" > $XDG_CONFIG_HOME/startkderc
+          '';
+        };
```

Comment:
This will delete all other options in startkderc, won't it? Also, setting `runUsingSystemd = false;` won't unset `systemdBoot`.

Reply:
It seems you can just call `${pkgs.plasma5Packages.kconfig}/bin/kwriteconfig5` directly in the script.

Reply:
This file isn't used for anything else, so overwriting it should be safe - albeit poor form, so it's been changed. And yes, it needs to run in the `false` case too to disable. Fixed.